### PR TITLE
Support adding Problem instances

### PR DESCRIPTION
**Pull request by tridao**

### CODE REVIEW

File: cvxpy/problems/problem.py [hunk 1]
```diff
@@ -404,3 +404,28 @@ def __str__(self):
     def __repr__(self):
         return "Problem(%s, %s)" % (repr(self.objective),
                                     repr(self.constraints))
+
+    def __add__(self, other):
+        if not isinstance(other, Problem):
+            raise TypeError("Can only add Problem instances.")
```
Comment (SteveDiamond) [inline cvxpy/problems/problem.py]: You should ``return NotImplemented`` instead of raising a TypeError here.

File: cvxpy/problems/problem.py [hunk 2]
```diff
@@ -404,3 +404,28 @@ def __str__(self):
...
+                           list(set(self.constraints + other.constraints)))
+        else:
+            raise TypeError(
+                "Can only add Problems that are all Minimize or Maximize.")
+
+    def __radd__(self, other):
+        if other == 0:
+            return self
+        else:
+            raise TypeError("Can only add Problem instances.")
```
Comment (SteveDiamond) [inline cvxpy/problems/problem.py]: You should return NotImplemented instead of raising a TypeError here.

Comment (tridao) [inline cvxpy/problems/problem.py]: You mean for both Min + Max and __radd__(), or just __radd__()?

Comment (SteveDiamond) [inline cvxpy/problems/problem.py]: Not for Min + Max. Just the two places I listed. You could raise a DCP
error instead of a TypeError for Min + Max.

**Pull request merged**